File: bomber/src/main/java/com/skx/tomike/bomber/basics/UrlEncodeActivity.java
package com.skx.tomike.bomber.basics;

import android.util.Log;

import com.skx.common.base.TitleConfig;
import com.skx.tomike.bomber.R;
import com.skx.common.base.BaseViewModel;
import com.skx.common.base.SkxBaseActivity;

import java.io.UnsupportedEncodingException;
import java.net.URLDecoder;
import java.net.URLEncoder;

public class UrlEncodeActivity extends SkxBaseActivity<BaseViewModel> {

    private static final String URL_ENCODE = "UTF-8";

    @Override
    protected void initParams() {

    }

    @Override
    protected TitleConfig configHeaderTitle() {
        return new TitleConfig.Builder().setTitleText("采用utf-8编码/解码").create();
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_url_encode;
    }

    @Override
    protected void initView() {
        String test1 = "12*8";
        String test2 = "556";
        String test3 = "abck.@";
        String test4 = "才发呢过@#￥%&*（）";


        try {
            Log.e("test1", URLEncoder.encode(test1, URL_ENCODE));
            Log.e("test2", URLEncoder.encode(test2, URL_ENCODE));
            Log.e("test3", URLEncoder.encode(test3, URL_ENCODE));
            Log.e("test4", URLEncoder.encode(test4, URL_ENCODE));
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
        }
    }
}

File: bomber/src/main/java/com/skx/tomike/bomber/basics/UrlEncodeActivity.kt
package com.skx.tomike.bomber.basics

import android.view.View
import android.widget.EditText
import android.widget.TextView
import com.alibaba.android.arouter.facade.annotation.Route
import com.skx.common.base.BaseViewModel
import com.skx.common.base.SkxBaseActivity
import com.skx.common.base.TitleConfig
import com.skx.tomike.bomber.R
import com.skx.tomike.bomber.ROUTE_PATH_URL_ENCODE
import java.net.URLDecoder
import java.net.URLEncoder

@Route(path = ROUTE_PATH_URL_ENCODE)
class UrlEncodeActivity : SkxBaseActivity<BaseViewModel>() {

    private val mEvOriginalVal: EditText by lazy {
        findViewById(R.id.et_urlEncode_original_value)
    }
    private val mTvResult: TextView by lazy {
        findViewById(R.id.tv_urlEncode_result)
    }

    override fun initParams() {}

    override fun configHeaderTitle(): TitleConfig {
        return TitleConfig.Builder().setTitleText("URL编码").create()
    }

    override fun layoutId(): Int {
        return R.layout.activity_url_encode
    }

    override fun initView() {
        findViewById<View>(R.id.tv_urlEncode_encoder).setOnClickListener {
            val oriVal = mEvOriginalVal.text.toString()
            val encodeResult: String? = try {
                URLEncoder.encode(oriVal, URL_ENCODE)
            } catch (e: Exception) {
                e.printStackTrace()
                e.message
            }
            mTvResult.text = encodeResult
        }
        findViewById<View>(R.id.tv_urlEncode_decoder).setOnClickListener {
            val oriVal = mEvOriginalVal.text.toString()
            val decodeResult: String? = try {
                URLDecoder.decode(oriVal, URL_ENCODE)
            } catch (e: Exception) {
                e.printStackTrace()
                e.message
            }
            mTvResult.text = decodeResult
        }
    }

    companion object {
        private const val URL_ENCODE = "UTF-8"
    }
}